### PR TITLE
Add openapi doc

### PR DESCRIPTION
**Pull request by tjquinno**
This adds a guide for using OpenAPI to the doc set.

The changes also add a card for the OpenAPI guide on the "Guides" overview page.

You can look at a rendered version of the guide by navigating to the `.adoc` file in my fork here: https://github.com/tjquinno/helidon/blob/add-openapi-doc/docs/src/main/docs/guides/07_openapi.adoc

It won't be identical to how it'd look on the web site but it's easier than reading the source.

### CODE REVIEW

File: docs/src/main/docs/guides/07_openapi.adoc [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+application's start-up.
+
+Add the link:{jandex-plugin-doc}[Jandex maven plug-in] to the `<build><plugins>` 
+section of your `pom.xml`:
+
+[source,xml]
+----
+<plugin>
+    <groupId>org.jboss.jandex</groupId>
+    <artifactId>jandex-maven-plugin</artifactId>
```
Comment (ljnelson) [inline docs/src/main/docs/guides/07_openapi.adoc]: Probably should have a `<version>` tag here, or include a `<pluginManagement>` stanza with the version specified.  I believe it must be 1.0.6 or later, yes? (Can't remember offhand.)

File: docs/src/main/docs/guides/07_openapi.adoc [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+
+==== Adding Helidon MP OpenAPI support to your app
+Add these two sections to the `<dependencies>` portion of your `pom.xml`:
+
+[source,xml,subs="attributes+"]
+----
+<dependency> <!--1-->
+    <groupId>org.eclipse.microprofile.openapi</groupId>
+    <artifactId>microprofile-openapi-api</artifactId>
+    <version>{microprofile-openapi-version}</version>
```
Comment (ljnelson) [inline docs/src/main/docs/guides/07_openapi.adoc]: Typo; need a `$` before the `{`

File: docs/src/main/docs/guides/07_openapi.adoc [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+</dependency>
+
+<dependency> <!--2-->
+    <groupId>io.helidon.microprofile.openapi</groupId>
+    <artifactId>helidon-microprofile-openapi</artifactId>
+    <version>{helidon-version}</version>
```
Comment (ljnelson) [inline docs/src/main/docs/guides/07_openapi.adoc]: Typo; need a `$` before the `{`

File: docs/src/main/docs/guides/07_openapi.adoc [hunk 4]
```diff
@@ -0,0 +1,240 @@
...
+<2> Adds the Helidon MP OpenAPI runtime support.
+
+== OpenAPI for Helidon SE
+Helidon SE does not process annotations. You can still use OpenAPI with your
+Helidon SE app by using a static OpenAPI document, a model reader,
+and a filter.
+
+To use OpenAPI from your Helidon SE app:
+
+1. in your `pom` add a dependency for Helidon SE support,
```
Comment (ljnelson) [inline docs/src/main/docs/guides/07_openapi.adoc]: Maybe `pom.xml` instead of just `pom`

File: docs/src/main/docs/guides/07_openapi.adoc [hunk 5]
```diff
@@ -0,0 +1,240 @@
...
+in `application.yaml` (see <<se_config, the SE config table>> below).
+
+Here is the additional dependency for Helidon SE OpenAPI runtime support:
+
+[source,xml,subs="attributes+"]
+----
+<dependency>
+    <groupId>io.helidon.openapi</groupId>
+    <artifactId>helidon-openapi</artifactId>
+    <version>{helidon-version}</version>
```
Comment (ljnelson) [inline docs/src/main/docs/guides/07_openapi.adoc]: Typo; need a `$` before the `{`

File: docs/src/main/docs/guides/07_openapi.adoc (hunk 1 above)
Comment (tjquinno) [reply docs/src/main/docs/guides/07_openapi.adoc]: Yes. I'll do that. Our example from which I drew all the code benefits from the inherited `<pluginManagement>` but a user won't be inheriting from our poms. Luckily it's pretty easy to pass the parameterized version we use in our build into the ADOC processing so when we build the site for a version of Helidon we capture the version we've used ourselves.

File: docs/src/main/docs/guides/07_openapi.adoc (hunk 2 above)
Comment (tjquinno) [reply docs/src/main/docs/guides/07_openapi.adoc]: Not really. Omitting the $ lets the AsciiDoctor processing substitute the attribute (property) assignment that's at the top of the `.adoc` file so the published guide actually has the real version in it.

File: docs/src/main/docs/guides/07_openapi.adoc (hunk 3 above)
Comment (tjquinno) [reply docs/src/main/docs/guides/07_openapi.adoc]: See earlier note about omitting the `$`.

File: docs/src/main/docs/guides/07_openapi.adoc (hunk 5 above)
Comment (tjquinno) [reply docs/src/main/docs/guides/07_openapi.adoc]: Same.

File: docs/src/main/docs/guides/07_openapi.adoc (hunk 4 above)
Comment (tjquinno) [reply docs/src/main/docs/guides/07_openapi.adoc]: Fixed.

**Comment by tjquinno** [comment]
In light of Joe's comment and a similar discussion in slack, I'm revising the write-ups into separate SE and MP documents. More pushes are in the works.

**Comment by tjquinno** [comment]
This recent push removes the draft single guide for SE and MP both and adds:

-  a new top-level nav item OpenAPI
-  a new SE-only page under that new nav item, and
-  a new MP-only page for OpenAPI under MicroProfile.

For ease of reading:
SE page: https://github.com/tjquinno/helidon/blob/add-openapi-doc/docs/src/main/docs/openapi/01_openapi.adoc

MP page: https://github.com/tjquinno/helidon/blob/add-openapi-doc/docs/src/main/docs/microprofile/08_openapi.adoc

**Comment by paulparkinson** [comment]
Looks great. I'll modify my demos to include this and the division of mp and and se is definitely good. Super small items: MP: Can you remove "(1)" and "(2)" from the pom.xml snippets (make them separate blocks if you want to represent they are separate dependency types)and use the fully qualified names for OASFilter. Same for OASModelReader in SE.

**Comment by tjquinno** [comment]
Hey, @paulparkinson. I've changed to use fully-qualified names for the OAS interfaces. Good point.

On your other suggestion -- My intent in using the callouts was to just explain why we add each of the two dependencies. That they have different scopes is related but incidental. 

Did you find the callouts distracting? Just plain not helpful? I'm trying to understand and address your underlying concern while still conveying in some way what I wanted to about the deps.

**Comment by paulparkinson** [comment]
I'm just a big fan of copy paste as-is , that's all. :) The callouts are helpful - could you just put them outside the gray/<dependency> blocks?

**Comment by tjquinno** [comment]
Oh oh oh, I get what you mean now.

That is a bug (I think) in GitHub's AsciiDoc rendering. The callout symbols are in XML comments in the `.adoc` source but when displayed on the GitHub pages the comment chars are gone, leaving the `(1)` and `(2)` out there unguarded.

Note that in the Java example the callout symbols are still behind the `//` comment introducer.

When I build the docs and run them locally, our rendered pages exclude the callout tags entirely from the copied text. 

I'll see if there's an open issue about this against GitHub's `.adoc` rendering.

**Comment by tjquinno** [comment]
I did not find an existing issue against GitHub's `.adoc` rendering, so I've created this one:

https://github.com/github/markup/issues/1276

**Comment by tjquinno** [comment]
Dan thought the GitHub/markup issue was better posted against Asciidoctor, so I closed the earlier issue and opened https://github.com/asciidoctor/asciidoctor/issues/3319

**Pull request merged**